### ISSUE TITLE
Cluster Rolling Update support the --ignore-daemonsets flag

### ISSUE THREAD

**Issue by denists1984** [kubernetes/kops#8391]
1. **What kops version are you running?** 
Version 1.15.0

2. **What Kubernetes version are you running?**
Version:"v1.15.6"

3. **What commands did you run? What is the simplest way to reproduce this issue?**
kops rolling-update cluster ${CLUSTER_NAME} --fail-on-validate-error="false" 

4. **What happened after the commands executed?**
The rolling update was started but in the logs i see that DaemonSets will be ignored and not evicted  during the node draining. 
`Draining the node: "fluentd-logging_node".`
`WARNING: ignoring DaemonSet-managed Pods: kube-system/fluentd-logging`

5. **What did you expect to happen?**
I would expect from `kops rolling-update cluster` to support the `--ignore-daemonsets=false` flag in order to drain the node completely and gracefully terminate the fluentd-logging pod before the node termination.

My question is if there are any plans to support this option in the future releases?

**Comment by mmerrill3** [comment]
+1

**Comment by mmerrill3** [comment]
It looks like that option is hard coded right now.  I have run into an issue with this setting with weave, since pods remain running on the node after it is deleted from the cluster, and weave is still participating in the ipam ring, after another weave instance claimed the IP space.

This is the location where the changes would need to allow for the option to ignore daemonsets would be included.

https://github.com/kubernetes/kops/blob/cba59afac45c1f20491ddb9f89133c951907ae19/pkg/instancegroups/instancegroups.go#L390-L391

I'm thinking that a good solution would be to allow this behavior at the instance group level, not universally across all the instance groups.

**Comment by johngmyers** [comment]
I suspect what might make sense is to first evict all non-daemonsets, then evict the daemonsets. But if, for example, fluentd-logging depended on weave then things would be more complicated.

/area rolling-update

**Comment by johngmyers** [comment]
On second thought, this might be a duplicate of #7119. It should probably be termination that gracefully stops the DaemonSet pods, not rolling update.

**Comment by mmerrill3** [comment]
With having weave running on a node that is cordoned, but not yet terminated, there is the chance for some portions of the IPAM ring to become unreachable.  It would be nice to have the option to really make sure nothing is running on the node before calling kubectl delete node, and then terminating the AWS instance.  This way, there's a guarantee that when the event for node deletion is processed by weave, there won't be a conflict between the "old" weave pod still running on the nodes that is being deleted from k8s, and another weave pod that will try to reclaim that IP space.  Weave-npc, on another peer in the IPAM ring, implements IP address reclamation when node deletion events occur.
To me, anecdotally, I don't care so much for any other daemonset running on the node that is being drained.  I just want the hammer option to ensure my network doesn't go haywire, affecting my entire cluster.  That hammer option is to also remove the daemon sets.

**Comment by mmerrill3** [comment]
@johngmyers what if we give users the ability to set the drain order via an annotation?  That way, we can put the network overlay pods last in the order of pods to evict?

**Comment by johngmyers** [comment]
My understanding of the use case was that you had a DaemonSet pod, `fluentd-logging` that had some local state that was desirable to flush before the instance gets terminated. That use case is not specific to rolling update; it applies to any instance termination.

If it's that your CNI can't handle the node being deleted before the DaemonSet pod is terminated, I have to wonder why you're using such a brittle CNI.

**Comment by mmerrill3** [comment]
Hi @johngmyers, thanks for the feedback.  I feel there's many use cases, beyond what CNI is used, where it may be preferable to also remove the daemon sets when we drain a node during a rolling-update.  The CNI example I gave is just one example.  Really, the original poster's issue, and my weave issue,  can both be solved by exposing "ingore-daemonsets", which is already inside the kubectl drain command to kops.  I started working on this, should have a PR this weekend.
